synapse/http/federation/matrix_federation_agent.py

```diff
@@ -116,7 +116,7 @@ def request(
         uri: bytes,
         headers: Optional[Headers] = None,
         bodyProducer: Optional[IBodyProducer] = None,
-    ) -> defer.Deferred:
+    ) -> Generator[defer.Deferred, Any, defer.Deferred]:
```

Comment:
We haven't switched this to `async` since `request` is part of the `IAgent` interface.